src/ape_console/_cli.py

```diff
@@ -1,23 +1,23 @@
 import faulthandler
 
-import click
 import IPython  # type: ignore
 
-from ape import networks
 from ape import project as default_project
-from ape.cli import network_option, verbose_option
+from ape.cli import network_bound_command, network_option, verbose_option
 from ape.version import version as ape_version  # type: ignore
 
 
-@click.command(short_help="Load the console", context_settings=dict(ignore_unknown_options=True))
+@network_bound_command(
+    short_help="Load the console",
+    context_settings=dict(ignore_unknown_options=True),
+)
```

Comment:
Ahhh, nvm my previous comment about `bind_network_command`. If I read this correctly, this overrides the use of `click.command`, correct?

Would there be another way to integrate this decorator that works with the context of the naked `click.command`?

I would say if not, we should look into bundling `click.command` into our CLI SDK so that plugin writers don't have to switch between both packages (and also allows us to refactor more effectively)

Reply:
Yes, so hence the past tense, re: overriding use of `click.command`

We could do something like that, like make an `ape_command` decorator. the `ApeGroup` was kind of along those lines, since it is just a normal `click.group` with the ability to create `bound_network_commands` if needed

Reply:
Hmm... I will have to think about this more. It seems like I am trying to _extend_ click here. I am trying to avoid abstracting all of click away because that may be just mostly passthroughs (like `click.echo`, `click.option` etc).
We could also go the reverse and requires everyone to import the class `NetworkBoundCommand` and set that via `cls=` on their `click.commands()`, which is fine. I just pushed a commit that makes it so they get the `network_option` for free though, so it would just require more setting up on the user end to make a network bound command.

Yeah I am going to think about this more. I am building out a custom ape plugin (`ape-jules`) to test that stuff.. It just makes JSON-RPC requests and prints stuff to the terminal, nothing fancy, but it is helpful in QA-ing this stuff.

Reply:
> Hmm... I will have to think about this more. It seems like I am trying to _extend_ click here. I am trying to avoid abstracting all of click away because that may be just mostly passthroughs (like `click.echo`, `click.option` etc). We could also go the reverse and require everyone to import the class `NetworkBoundCommand` and set that via `cls=` on their `click.commands()`, which is fine. I just pushed a commit that makes it so they get the `network_option` for free though, so it would just require more setting up on the user end to make a network bound command.

Yes, seems like we have a choice here to fully wrap click in everything we're doing, or be more transparent and let click do more of the work. I'm okay with either choice, but we need to commit to one. We've been sort of trending down this path towards wrapping everything click into a "CLI SDK" for plugin writers

---

> Yeah I am going to think about this more. I am building out a custom ape plugin (`ape-jules`) to test that stuff.. It just makes JSON-RPC requests and prints stuff to the terminal, nothing fancy, but it is helpful in QA-ing this stuff.

Consider making this a permanent addition to the test suite, e.g. `ape-dummy`, so we have a way to test integration of a new plugin for all the features that are not tested via the 1st class plugins already in the repo